### PR TITLE
Feature/ml 426 ml layers to source

### PR DESCRIPTION
**Pull request by AlexTeslenko**
ml 426  Moved ml/layers to source

### CODE REVIEW

File: libs/ml/include/ml/layers/skip_gram.hpp [hunk 1]
```diff
@@ -18,15 +18,11 @@
...
 #include "ml/ops/embeddings.hpp"
-#include "ml/ops/flatten.hpp"
-#include "ml/ops/matrix_multiply.hpp"
 #include "ml/ops/placeholder.hpp"
-#include "ml/ops/transpose.hpp"
 
-#include <cmath>
-#include <random>
+//#include <cmath>
+//#include <random>
```
Comment (vhavrylov-ncube) [inline libs/ml/include/ml/layers/skip_gram.hpp]: pls remove this

File: libs/ml/src/layers/convolution_1d.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+///////////////////////////////
+/// EXPLICIT INSTANTIATIONS ///
+///////////////////////////////
+
+// template class Convolution1D<math::Tensor<int8_t>>;
+// template class Convolution1D<math::Tensor<int16_t>>;
+template class Convolution1D<math::Tensor<int32_t>>;
+template class Convolution1D<math::Tensor<int64_t>>;
+// template class Convolution1D<math::Tensor<uint8_t>>;
+// template class Convolution1D<math::Tensor<uint16_t>>;
```
Comment (vhavrylov-ncube) [inline libs/ml/src/layers/convolution_1d.cpp]: are these commented instantiations not needed?

Comment (AlexTeslenko) [reply libs/ml/src/layers/convolution_1d.cpp]: they don't work for now, but should work in future

File: libs/ml/include/ml/layers/PRelu.hpp [hunk 3]
```diff
@@ -17,16 +17,8 @@
 //
 //------------------------------------------------------------------------------
 
-#include "math/tensor/tensor.hpp"
 #include "ml/layers/fully_connected.hpp"
```
Comment (baykaner) [inline libs/ml/include/ml/layers/PRelu.hpp]: move include into source

File: libs/ml/include/ml/layers/convolution_1d.hpp [hunk 4]
```diff
@@ -18,17 +18,9 @@
 //------------------------------------------------------------------------------
 
 #include "ml/core/subgraph.hpp"
```
Comment (baykaner) [inline libs/ml/include/ml/layers/convolution_1d.hpp:20]: move includes into source across all layers

File: libs/ml/src/ops/activation.cpp [hunk 5]
```diff
@@ -65,7 +65,30 @@ std::string AddActivationNode(ActivationType type, Graph<T> *g, std::string name
...
+template std::string AddActivationNode<math::Tensor<int16_t>>(ActivationType                type,
+                                                              Graph<math::Tensor<int16_t>> *g,
+                                                              std::string name, std::string input);
+template std::string AddActivationNode<math::Tensor<int32_t>>(ActivationType                type,
+                                                              Graph<math::Tensor<int32_t>> *g,
+                                                              std::string name, std::string input);
+template std::string AddActivationNode<math::Tensor<int64_t>>(ActivationType                type,
+                                                              Graph<math::Tensor<int64_t>> *g,
+                                                              std::string name, std::string input);
+template std::string AddActivationNode<math::Tensor<uint8_t>>(ActivationType                type,
```
Comment (baykaner) [inline libs/ml/src/ops/activation.cpp]: lets not support the uints anymore

**Review by baykaner — CHANGES_REQUESTED** [review]

File: libs/ml/include/ml/layers/convolution_1d.hpp (hunk 4 above)
Comment (AlexTeslenko) [reply libs/ml/include/ml/layers/convolution_1d.hpp:20]: but I need these include to resolve namespaces:

using ArrayPtrType  = std::shared_ptr<TensorType>;
  using SizeType      = fetch::math::SizeType;
  using WeightsInit   = fetch::ml::ops::WeightsInitialisation;

Comment (AlexTeslenko) [reply libs/ml/include/ml/layers/convolution_1d.hpp:20]: I tried to forward declare subgraph, WeightsInitialisation, LayerConvolution1DSaveableParams, Activation etc,  but they require definition in header:

Convolution1D(SizeType const output_channels, SizeType const input_channels,
                SizeType const kernel_size, SizeType const stride_size,
                details::ActivationType const activation_type = details::ActivationType::NOTHING,
                std::string const &           name            = "Conv1D",
                WeightsInit const             init_mode       = WeightsInit::XAVIER_GLOROT,
                SizeType const                seed            = 123456789);

**Review by baykaner — DISMISSED** [review]

**Review by vhavrylov-ncube — DISMISSED** [review]
looks good to me

**Review by vhavrylov-ncube — APPROVED** [review]
LGTM

**Pull request merged**